### ISSUE TITLE
unreliable when trying to install via docker / docker compose

### ISSUE THREAD

**Issue by WolfgangFahl** [papermerge/papermerge-core#440]
When trying papermerge i get errors such as the one below or no answer at all. The software seem to be not very robust and make assumptions that are not valid when still tinkering with multiple versions and approaches as are scattered over different sources of information. I tried e.g.
* https://docs.papermerge.io/3.2/setup/docker-compose/
but also found
* https://hub.docker.com/r/papermerge/papermerge
see also
* https://www.reddit.com/r/Papermerge/comments/1ez7n06/papermount_making_inbox_and_db_available_via_mount/

Effectively papermerge is not accessible for me yet due to these startup problems.

1" 500 177 "http://arche.bitplan.com:8008/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:129.0) Gecko/20100101 Firefox/129.0"
web-1     | Exception in ASGI application
web-1     | Traceback (most recent call last):
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
web-1     |     result = await app(  # type: ignore[func-returns-value]
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
web-1     |     return await self.app(scope, receive, send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
web-1     |     raise exc from None
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
web-1     |     await self.app(scope, inner_receive, inner_send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
web-1     |     await super().__call__(scope, receive, send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
web-1     |     await self.middleware_stack(scope, receive, send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
web-1     |     raise exc
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
web-1     |     await self.app(scope, receive, _send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
web-1     |     raise exc
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
web-1     |     await self.app(scope, receive, sender)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
web-1     |     await route.handle(scope, receive, send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
web-1     |     await self.app(scope, receive, send)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
web-1     |     response = await func(request)
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 299, in app
web-1     |     raise e
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 294, in app
web-1     |     raw_response = await run_endpoint_function(
web-1     |   File "/auth_server_app/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
web-1     |     return await dependant.call(**values)
web-1     |   File "/auth_server_app/auth_server/main.py", line 148, in verify_endpoint
web-1     |     user = db.get_user_uuid(session, UUID(user_id))
web-1     |   File "/usr/local/lib/python3.10/uuid.py", line 177, in __init__
web-1     |     raise ValueError('badly formed hexadecimal UUID string')
web-1     | ValueError: badly formed hexadecimal UUID string

**Comment by ciur** [comment]
Please describe you local setup, which assumptions are not true in your case, what version you tried to install, step by step of you setup process.

**Comment by WolfgangFahl** [comment]
I am using 
```bash
#!/bin/bash
# WF 2024-08-24
# Setup and manage Papermerge using Docker Compose
VERSION="0.0.2"
pm_root=$HOME/.papermerge
pm_compose=$pm_root/docker-compose.yml
pm_env=$pm_root/.env

# Color definitions
blue='\033[0;34m'
red='\033[0;31m'
green='\033[0;32m'
endColor='\033[0m'

# Function to display colored messages
color_msg() {
  local l_color="$1"
  local l_msg="$2"
  echo -e "${l_color}$l_msg${endColor}"
}

# Function to display errors
error() {
  local l_msg="$1"
  color_msg $red "Error:" 1>&2
  color_msg $red "\t$l_msg" 1>&2
  exit 1
}

# Function to display negative messages
negative() {
  local l_msg="$1"
  color_msg $red "❌:$l_msg"
}

# Function to display positive messages
positive() {
  local l_msg="$1"
  color_msg $green "✅:$l_msg"
}

# Function to display usage information
usage() {
  echo "Usage: $0 [OPTIONS]"
  echo "Options:"
  echo "  -h, --help             Show this help message"
  echo "  -s, --setup            Setup Papermerge Docker Compose configuration"
  echo "  -u, --up               Start Papermerge services"
  echo "  -dn, --down            Stop Papermerge services"
  echo "  -d, --debug            Enable debug mode"
  echo "  -l, --logs             View Papermerge logs"
  echo "  -c, --config           View current configuration"
  echo "  -v, --version          Show version information"
  exit 1
}

# Function to setup Papermerge
setup_papermerge() {
  if [ ! -d "$pm_root" ]; then
    color_msg $blue "Creating $pm_root"
    mkdir -p "$pm_root"
  fi

  if [ ! -f "$pm_compose" ]; then
    cat << EOF > "$pm_compose"
x-backend: &common
  image: papermerge/papermerge:3.2
  env_file: .env
  volumes:
    - data:/db
    - index_db:/core_app/index_db
    - media:/core_app/media
services:
  web:
    <<: *common
    ports:
     - "8008:80"
    depends_on:
      - redis
  worker:
    <<: *common
    command: worker
  redis:
    image: redis:6
volumes:
  data:
  index_db:
  media:
EOF
  fi

  if [ ! -f "$pm_env" ]; then
    error ".env file not found. Please create $pm_env with the necessary environment variables."
  else
    positive "Papermerge Docker Compose configuration has been set up in $pm_root"
  fi
}

# Function to start Papermerge services
start_services() {
  if [ ! -f "$pm_env" ]; then
    error ".env file not found. Please create $pm_env with the necessary environment variables."
  fi
  cd "$pm_root" && docker compose up
  positive "Papermerge services started"
}

# Function to stop Papermerge services
stop_services() {
  cd "$pm_root" && docker compose down
  positive "Papermerge services stopped"
}

# Function to view Papermerge logs
view_logs() {
  cd "$pm_root" && docker compose logs -f
}

# Function to view current configuration
view_config() {
  if [ -f "$pm_compose" ]; then
    cat "$pm_compose"
  else
    negative "docker-compose.yml not found. Please run setup first."
  fi
}

# Parse command line arguments
while [[ $# -gt 0 ]]; do
  case $1 in
    -h|--help) usage ;;
    -s|--setup) setup_papermerge ;;
    -u|--up) start_services ;;
    -d|--debug) set -x;;
    -dn|--down) stop_services ;;
    -l|--logs) view_logs ;;
    -c|--config) view_config ;;
    -v|--version) echo "Version: $VERSION"; exit 0 ;;
    *) error "Unknown option: $1" ;;
  esac
  shift
done

# If no arguments provided, show usage
if [ $# -eq 0 ]; then
  usage
fi
```
**usage**
```
papermerge -h
Usage: /home/wf/bin/papermerge [OPTIONS]
Options:
  -h, --help             Show this help message
  -s, --setup            Setup Papermerge Docker Compose configuration
  -u, --up               Start Papermerge services
  -dn, --down            Stop Papermerge services
  -d, --debug            Enable debug mode
  -l, --logs             View Papermerge logs
  -c, --config           View current configuration
  -v, --version          Show version information
```
papermerge -s
✅:Papermerge Docker Compose configuration has been set up in /home/wf/.papermerge/
```bash
cat /home/wf/.papermerge/docker-compose.yml 
version: "3.9"

x-backend: &common
  image: papermerge/papermerge:3.2
  environment:
    PAPERMERGE__SECURITY__SECRET_KEY: QWERTY_67890
    PAPERMERGE__AUTH__USERNAME: admin
    PAPERMERGE__AUTH__PASSWORD: LsbF%2v1
    PAPERMERGE__REDIS__URL: redis://redis:6379/0
  volumes:
    - data:/db
    - index_db:/core_app/index_db
    - media:/core_app/media

services:
  web:
    <<: *common
    ports:
     - "8008:80"
    depends_on:
      - redis
  worker:
    <<: *common
    command: worker
  redis:
    image: redis:6

volumes:
  data:
  index_db:
  media:
```

papermerge -u
```
WARN[0000] /home/wf/.papermerge/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
[+] Running 3/0
 ✔ Container papermerge-redis-1   Create...                                0.0s 
 ✔ Container papermerge-worker-1  Creat...                                 0.0s 
 ✔ Container papermerge-web-1     Created                                  0.0s 
Attaching to redis-1, web-1, worker-1
redis-1   | 1:C 24 Aug 2024 10:40:53.985 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-1   | 1:C 24 Aug 2024 10:40:53.985 # Redis version=6.2.14, bits=64, commit=00000000, modified=0, pid=1, just started
redis-1   | 1:C 24 Aug 2024 10:40:53.985 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-1   | 1:M 24 Aug 2024 10:40:53.986 * monotonic clock: POSIX clock_gettime
redis-1   | 1:M 24 Aug 2024 10:40:53.986 * Running mode=standalone, port=6379.
redis-1   | 1:M 24 Aug 2024 10:40:53.986 # Server initialized
redis-1   | 1:M 24 Aug 2024 10:40:53.986 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis-1   | 1:M 24 Aug 2024 10:40:53.987 * Ready to accept connections
worker-1  | Installing dependencies from lock file
web-1     | Installing dependencies from lock file
worker-1  | 
worker-1  | No dependencies to install or update
worker-1  | 
worker-1  | Installing the current project: papermerge-core (3.2)
web-1     | 
web-1     | No dependencies to install or update
web-1     | 
web-1     | Installing the current project: papermerge-core (3.2)
worker-1  | Operations to perform:
worker-1  |   Apply all migrations: auth, contenttypes, core, sites
worker-1  | Running migrations:
worker-1  |   No migrations to apply.
web-1     | Operations to perform:
web-1     |   Apply all migrations: auth, contenttypes, core, sites
web-1     | Running migrations:
web-1     |   No migrations to apply.
worker-1  | Installing dependencies from lock file
web-1     | Installing dependencies from lock file
worker-1  | 
worker-1  | No dependencies to install or update
web-1     | 
web-1     | No dependencies to install or update
web-1     | 
web-1     | Installing the current project: auth-server (0.9.0)
worker-1  | 
worker-1  | Installing the current project: auth-server (0.9.0)
worker-1  | User 'admin' already exists.
web-1     | User 'admin' already exists.
worker-1  | /core_app/.venv/lib/python3.10/site-packages/celery/platforms.py:829: SecurityWarning: You're running the worker with superuser privileges: this is
worker-1  | absolutely not recommended!
worker-1  | 
worker-1  | Please specify a different user using the --uid option.
worker-1  | 
worker-1  | User information: uid=0 euid=0 gid=0 egid=0
worker-1  | 
worker-1  |   warnings.warn(SecurityWarning(ROOT_DISCOURAGED.format(
worker-1  |  
worker-1  |  -------------- worker-node-9d8e9f861234@papermerge v5.3.6 (emerald-rush)
worker-1  | --- ***** ----- 
worker-1  | -- ******* ---- Linux-6.8.0-40-generic-x86_64-with-glibc2.36 2024-08-24 12:41:03
worker-1  | - *** --- * --- 
worker-1  | - ** ---------- [config]
worker-1  | - ** ---------- .> app:         papermerge:0x7b0b1b211e10
worker-1  | - ** ---------- .> transport:   redis://redis:6379/0
worker-1  | - ** ---------- .> results:     disabled://
worker-1  | - *** --- * --- .> concurrency: 4 (prefork)
worker-1  | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
worker-1  | --- ***** ----- 
worker-1  |  -------------- [queues]
worker-1  |                 .> celery           exchange=papermerge(direct) key=papermerge
worker-1  |                 
worker-1  | 
worker-1  | [2024-08-24 12:41:04,051: WARNING/MainProcess] /core_app/.venv/lib/python3.10/site-packages/celery/worker/consumer/consumer.py:507: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
worker-1  | whether broker connection retries are made during startup in Celery 6.0 and above.
worker-1  | If you wish to retain the existing behavior for retrying connections on startup,
worker-1  | you should set broker_connection_retry_on_startup to True.
worker-1  |   warnings.warn(
worker-1  | 
worker-1  | [2024-08-24 12:41:04,060: WARNING/MainProcess] /core_app/.venv/lib/python3.10/site-packages/celery/worker/consumer/consumer.py:507: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
worker-1  | whether broker connection retries are made during startup in Celery 6.0 and above.
worker-1  | If you wish to retain the existing behavior for retrying connections on startup,
worker-1  | you should set broker_connection_retry_on_startup to True.
worker-1  |   warnings.warn(
worker-1  | 
web-1     | 2024-08-24 10:41:05,104 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
web-1     | 2024-08-24 10:41:05,106 INFO supervisord started with pid 1
web-1     | 2024-08-24 10:41:06,108 INFO spawned: 'auth_server' with pid 64
web-1     | 2024-08-24 10:41:06,110 INFO spawned: 'core' with pid 65
web-1     | 2024-08-24 10:41:06,112 INFO spawned: 'nginx' with pid 66
web-1     | 2024/08/24 10:41:06 [notice] 66#66: using the "epoll" event method
web-1     | 2024/08/24 10:41:06 [notice] 66#66: nginx/1.22.1
web-1     | 2024/08/24 10:41:06 [notice] 66#66: OS: Linux 6.8.0-40-generic
web-1     | 2024/08/24 10:41:06 [notice] 66#66: getrlimit(RLIMIT_NOFILE): 1048576:1048576
web-1     | 2024/08/24 10:41:06 [notice] 66#66: start worker processes
web-1     | 2024/08/24 10:41:06 [notice] 66#66: start worker process 67
web-1     | 2024-08-24 10:41:07,126 INFO success: auth_server entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
web-1     | 2024-08-24 10:41:07,127 INFO success: core entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
web-1     | 2024-08-24 10:41:07,127 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
```

i can then login per the credentials of my .env file
```
PAPERMERGE__SECURITY__SECRET_KEY=***SecretKey***
PAPERMERGE__AUTH__USERNAME=admin
PAPERMERGE__AUTH__PASSWORD=***SecretPassword***
```
<img width="823" alt="grafik" src="https://github.com/user-attachments/assets/1cd7ea01-e0c5-40ab-97bc-58c63355b463">

and can initially see the inbox

<img width="1005" alt="grafik" src="https://github.com/user-attachments/assets/d4932c48-dd4b-4353-b939-bc44d990d9e5">

On my attempts the last few days more often than not i would end up with errors like the one reported here. Let's see how it goes this time. Still i think it is a good idea to make the code more robust to avoid blank screens when an internal erro like this happens.

**Comment by spam0115** [comment]
I agree with OP.  I am running your premade docker image on Windows docker and I just get the following error message when I browse to the image's url: {"detail":"Not Found"}.  Running a GET or POST on http://localhost:8000/api/auth/login/ produces the same error.

**Comment by WolfgangFahl** [comment]
@spam0115  currently this is exactly the state of one of my machines {"detail":"Not Found"}
The docker logs seem to have no proper hint what the issue is:
```bash
papermerge -l
papermerge_redis   | 1:C 25 Aug 2024 06:11:42.823 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
papermerge_redis   | 1:C 25 Aug 2024 06:11:42.824 # Redis version=6.2.14, bits=64, commit=00000000, modified=0, pid=1, just started
papermerge_redis   | 1:C 25 Aug 2024 06:11:42.824 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
papermerge_worker  | Installing dependencies from lock file
papermerge_worker  | 
papermerge_worker  | No dependencies to install or update
papermerge_worker  | 
papermerge_worker  | Installing the current project: papermerge-core (3.0.3)
papermerge_worker  | Operations to perform:
papermerge_worker  |   Apply all migrations: auth, contenttypes, core, sites
papermerge_worker  | Running migrations:
papermerge_worker  |   No migrations to apply.
papermerge_worker  | Installing dependencies from lock file
papermerge_worker  | 
papermerge_worker  | Package operations: 6 installs, 0 updates, 0 removals
papermerge_worker  | 
papermerge_worker  |   • Installing iniconfig (2.0.0)
papermerge_worker  |   • Installing packaging (23.2)
papermerge_worker  |   • Installing pluggy (1.3.0)
papermerge_worker  |   • Installing tomli (2.0.1)
papermerge_worker  |   • Installing pytest (7.4.4)
papermerge_worker  |   • Installing pytest-env (0.8.2)
papermerge_worker  | 
papermerge_worker  | Installing the current project: auth-server (0.6.4)
papermerge_worker  | User 'admin' already exists.
papermerge_worker  | /core_app/.venv/lib/python3.10/site-packages/celery/platforms.py:829: SecurityWarning: You're running the worker with superuser privileges: this is
papermerge_worker  | absolutely not recommended!
papermerge_worker  | 
papermerge_worker  | Please specify a different user using the --uid option.
papermerge_worker  | 
papermerge_worker  | User information: uid=0 euid=0 gid=0 egid=0
papermerge_worker  | 
papermerge_worker  |   warnings.warn(SecurityWarning(ROOT_DISCOURAGED.format(
papermerge_worker  |  
papermerge_worker  |  -------------- worker-node-b85aca335454@papermerge v5.3.6 (emerald-rush)
papermerge_worker  | --- ***** ----- 
papermerge_worker  | -- ******* ---- Linux-6.8.0-40-generic-x86_64-with-glibc2.36 2024-08-25 08:12:05
papermerge_worker  | - *** --- * --- 
papermerge_worker  | - ** ---------- [config]
papermerge_worker  | - ** ---------- .> app:         papermerge:0x71dca3a32110
papermerge_worker  | - ** ---------- .> transport:   memory://localhost//
papermerge_worker  | - ** ---------- .> results:     disabled://
papermerge_worker  | - *** --- * --- .> concurrency: 4 (prefork)
papermerge_worker  | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
papermerge_worker  | --- ***** ----- 
papermerge_worker  |  -------------- [queues]
papermerge_worker  |                 .> celery           exchange=papermerge(direct) key=papermerge
papermerge_worker  |                 
papermerge_web     | Installing dependencies from lock file
papermerge_web     | 
papermerge_worker  | 
papermerge_worker  | [2024-08-25 08:12:05,421: WARNING/MainProcess] /core_app/.venv/lib/python3.10/site-packages/celery/worker/consumer/consumer.py:507: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
papermerge_worker  | whether broker connection retries are made during startup in Celery 6.0 and above.
papermerge_worker  | If you wish to retain the existing behavior for retrying connections on startup,
papermerge_worker  | you should set broker_connection_retry_on_startup to True.
papermerge_worker  |   warnings.warn(
papermerge_worker  | 
papermerge_worker  | [2024-08-25 08:12:05,426: WARNING/MainProcess] /core_app/.venv/lib/python3.10/site-packages/celery/worker/consumer/consumer.py:507: CPendingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine
papermerge_worker  | whether broker connection retries are made during startup in Celery 6.0 and above.
papermerge_worker  | If you wish to retain the existing behavior for retrying connections on startup,
papermerge_worker  | you should set broker_connection_retry_on_startup to True.
papermerge_worker  |   warnings.warn(
papermerge_worker  | 
papermerge_redis   | 1:M 25 Aug 2024 06:11:42.824 * monotonic clock: POSIX clock_gettime
papermerge_redis   | 1:M 25 Aug 2024 06:11:42.825 * Running mode=standalone, port=6379.
papermerge_redis   | 1:M 25 Aug 2024 06:11:42.825 # Server initialized
papermerge_web     | No dependencies to install or update
papermerge_redis   | 1:M 25 Aug 2024 06:11:42.825 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
papermerge_redis   | 1:M 25 Aug 2024 06:11:42.825 * Ready to accept connections
papermerge_web     | 
papermerge_web     | Installing the current project: papermerge-core (3.0.3)
papermerge_web     | Operations to perform:
papermerge_web     |   Apply all migrations: auth, contenttypes, core, sites
papermerge_web     | Running migrations:
papermerge_web     |   No migrations to apply.
papermerge_web     | Installing dependencies from lock file
papermerge_web     | 
papermerge_web     | Package operations: 6 installs, 0 updates, 0 removals
papermerge_web     | 
papermerge_web     |   • Installing iniconfig (2.0.0)
papermerge_web     |   • Installing packaging (23.2)
papermerge_web     |   • Installing pluggy (1.3.0)
papermerge_web     |   • Installing tomli (2.0.1)
papermerge_web     |   • Installing pytest (7.4.4)
papermerge_web     |   • Installing pytest-env (0.8.2)
papermerge_web     | 
papermerge_web     | Installing the current project: auth-server (0.6.4)
papermerge_web     | User 'admin' already exists.
papermerge_web     | /usr/lib/python3/dist-packages/supervisor/options.py:474: UserWarning: Supervisord is running as root and it is searching for its configuration file in default locations (including its current working directory); you probably want to specify a "-c" argument specifying an absolute path to a configuration file for improved security.
papermerge_web     |   self.warnings.warn(
papermerge_web     | 2024-08-25 06:12:09,776 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
papermerge_web     | 2024-08-25 06:12:09,787 INFO supervisord started with pid 1
papermerge_web     | 2024-08-25 06:12:10,790 INFO spawned: 'tivoli' with pid 81
papermerge_web     | 2024-08-25 06:12:10,792 INFO spawned: 'auth_server' with pid 82
papermerge_web     | 2024-08-25 06:12:10,794 INFO spawned: 'core' with pid 83
papermerge_web     | 2024-08-25 06:12:10,798 INFO spawned: 'nginx' with pid 84
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: using the "epoll" event method
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: nginx/1.22.1
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: OS: Linux 6.8.0-40-generic
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: getrlimit(RLIMIT_NOFILE): 1048576:1048576
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: start worker processes
papermerge_web     | 2024/08/25 06:12:10 [notice] 84#84: start worker process 85
papermerge_web     | 2024-08-25 06:12:12,318 INFO success: tivoli entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
papermerge_web     | 2024-08-25 06:12:12,318 INFO success: auth_server entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
papermerge_web     | 2024-08-25 06:12:12,319 INFO success: core entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
papermerge_web     | 2024-08-25 06:12:12,319 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
papermerge_web     | INFO:     Uvicorn running on http://0.0.0.0:8100 (Press CTRL+C to quit)
papermerge_web     | INFO:     Started parent process [81]
papermerge_web     | INFO:     Started server process [95]
papermerge_web     | INFO:     Waiting for application startup.
papermerge_web     | INFO:     Application startup complete.
papermerge_web     | INFO:     Started server process [96]
papermerge_web     | INFO:     Waiting for application startup.
papermerge_web     | INFO:     Started server process [94]
papermerge_web     | INFO:     Waiting for application startup.
papermerge_web     | INFO:     Application startup complete.
papermerge_web     | INFO:     Application startup complete.
```
inside the container 
```bash
papermerge -b
ps xww
    PID TTY      STAT   TIME COMMAND
      1 ?        Ss     0:04 /usr/bin/python3 /usr/bin/supervisord
     73 pts/0    Ss+    0:00 /bin/bash /run_sqlite_web.sh
     81 ?        S      0:00 /tivoli_app/.venv/bin/python /tivoli_app/.venv/bin/uvicorn tivoli.main:app --workers 3 --host 0.0.0.0 --port 8100
     82 ?        S      0:00 /auth_server_app/.venv/bin/python /auth_server_app/.venv/bin/uvicorn auth_server.main:app --workers 3 --host 0.0.0.0 --port 4010 --log-config /core_app/logging.yml
     83 ?        S      0:00 /core_app/.venv/bin/python /core_app/.venv/bin/uvicorn config.asgi:fastapp --workers 4 --host 0.0.0.0 --app-dir /core_app/ --log-config /core_app/logging.yml --port 8000
     84 ?        S      0:00 nginx: master process /usr/sbin/nginx
     93 ?        S      0:00 /tivoli_app/.venv/bin/python -c from multiprocessing.resource_tracker import main;main(4)
     94 ?        S      0:45 /tivoli_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=7) --multiprocessing-fork
     95 ?        S      0:45 /tivoli_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=9) --multiprocessing-fork
     96 ?        S      0:45 /tivoli_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=11) --multiprocessing-fork
     97 ?        S      0:00 /auth_server_app/.venv/bin/python -c from multiprocessing.resource_tracker import main;main(4)
     98 ?        S      0:44 /auth_server_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=7) --multiprocessing-fork
     99 ?        S      0:44 /auth_server_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=9) --multiprocessing-fork
    100 ?        S      0:44 /auth_server_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=11) --multiprocessing-fork
    102 ?        S      0:00 /core_app/.venv/bin/python -c from multiprocessing.resource_tracker import main;main(4)
    103 ?        S      0:45 /core_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=7) --multiprocessing-fork
    104 ?        S      0:45 /core_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=9) --multiprocessing-fork
    105 ?        S      0:45 /core_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=11) --multiprocessing-fork
    106 ?        S      0:45 /core_app/.venv/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=13) --multiprocessing-fork
    156 pts/0    S+     0:03 /usr/local/bin/python /usr/local/bin/sqlite_web --host 0.0.0.0 --port 8080 /db/db.sqlite3
    210 pts/1    Ss     0:00 /bin/bash
    217 pts/1    R+     0:00 ps xww
 ```
 
 that is all greek to me

**Comment by marfoerst** [comment]
Backlog triage: closing this one out.

This has been open since 2024 without the additional detail that was requested, and the installation flow has changed substantially across the 3.x releases since then.

If you are still blocked on a current release, please open a new issue with your compose file and the container logs and we will help you get it running. Thanks for taking the time to report it originally.
